File: src/main/java/me/udnek/fnafu/player/PlayerContainer.java
package me.udnek.fnafu.player;

import me.udnek.fnafu.player.type.Animatronic;
import me.udnek.fnafu.player.type.Survivor;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public class PlayerContainer {

    private final List<Survivor> survivors = new ArrayList<>();
    private final List<Animatronic> animatronics = new ArrayList<>();

    public Survivor getSurvivor(Player player){
        for (Survivor survivor : survivors) {
            if (survivor.getPlayer() == player) return survivor;
        }
        return null;
    }
    public Animatronic getAnimatronic(Player player){
        for (Animatronic animatronic : animatronics) {
            if (animatronic.getPlayer() == player) return animatronic;
        }
        return null;
    }

    public void add(Survivor survivor){
        survivors.add(survivor);
    }
    public void add(Animatronic animatronic){
        animatronics.add(animatronic);
    }
    public boolean remove(Player player){
        List<Survivor> survivorList = getSurvivors(false);
        for (int i = 0; i < survivorList.size(); i++) {
            Survivor survivor = survivorList.get(i);
            if (survivor.getPlayer() == player) {
                survivorList.remove(i);
                return true;
            }
        }
        List<Animatronic> animatronicList = getAnimatronics(false);
        for (int i = 0; i < animatronicList.size(); i++) {
            Animatronic animatronic = animatronicList.get(i);
            if (animatronic.getPlayer() == player) {
                animatronicList.remove(i);
                return true;
            }
        }
        return false;
    }

    public List<Survivor> getSurvivors(boolean returnCopy) {
        if (returnCopy) return new ArrayList<>(survivors);
        return survivors;
    }
    public List<Animatronic> getAnimatronics(boolean returnCopy) {
        if (returnCopy) return new ArrayList<>(animatronics);
        return animatronics;
    }
    public List<FnafUPlayer> getAll(){
        ArrayList<FnafUPlayer> fnafUPlayers = new ArrayList<>();
        fnafUPlayers.addAll(survivors);
        fnafUPlayers.addAll(animatronics);
        return fnafUPlayers;
    }
    public boolean contains(Player player){
        return (containsSurvivor(player) || containsAnimatronic(player));

    }

    public int getAliveSurvivorsAmount(){
        int amount = 0;
        for (Survivor survivor : survivors) {
            if (survivor.getHealthState().isAlive()){
                amount++;
            }
        }
        return amount;
    }

    public boolean containsSurvivor(Player player){
        for (Survivor survivor : survivors) {
            if (survivor.getPlayer() == player) return true;
        }
        return false;
    }
    public boolean containsAnimatronic(Player player){
        for (Animatronic animatronic : animatronics) {
            if (animatronic.getPlayer() == player) return true;
        }
        return false;
    }
}

File: src/main/java/me/udnek/fnafu/player/PlayerContainer.kt
package me.udnek.fnafu.player

import org.bukkit.entity.Player


class PlayerContainer {
    private val survivors: MutableList<FnafUPlayer> = ArrayList()
    private val animatronics: MutableList<FnafUPlayer> = ArrayList()

    val all: List<FnafUPlayer>
        get() {
            val fnafUPlayers = ArrayList<FnafUPlayer>()
            fnafUPlayers.addAll(survivors)
            fnafUPlayers.addAll(animatronics)
            return fnafUPlayers
        }

    val aliveSurvivorsAmount: Int
        get() {
            return survivors.size
        }


    fun getSurvivor(player: Player): FnafUPlayer? {
        return survivors.firstOrNull { it.player == player }
    }

    fun getAnimatronic(player: Player): FnafUPlayer? {
        return animatronics.firstOrNull { it.player == player }
    }


    fun getPlayer(player: Player): FnafUPlayer? {
        val survivor = getSurvivor(player)
        if (survivor != null) return survivor
        return getAnimatronic(player)
    }

    fun addSurvivor(survivor: FnafUPlayer): Boolean {
        if (containsSurvivor(survivor.player)) return false
        return survivors.add(survivor)
    }

    fun addAnimatronic(animatronic: FnafUPlayer): Boolean {
        if (containsAnimatronic(animatronic.player)) return false
        return animatronics.add(animatronic)
    }

    fun remove(player: Player): Boolean {
        val survivorList = survivors
        for (i in survivorList.indices) {
            val survivor = survivorList[i]
            if (survivor.player == player) {
                survivorList.removeAt(i)
                return true
            }
        }
        val animatronicList = animatronics
        for (i in animatronicList.indices) {
            val animatronic = animatronicList[i]
            if (animatronic.player == player) {
                animatronicList.removeAt(i)
                return true
            }
        }
        return false
    }

    fun getSurvivors(returnCopy: Boolean): MutableList<FnafUPlayer> {
        if (returnCopy) return ArrayList(survivors)
        return survivors
    }

    fun getAnimatronics(returnCopy: Boolean): MutableList<FnafUPlayer> {
        if (returnCopy) return ArrayList(animatronics)
        return animatronics
    }


    fun contains(player: Player): Boolean = containsSurvivor(player) || containsAnimatronic(player)

    fun containsSurvivor(player: Player): Boolean = survivors.any { it.player == player }

    fun containsAnimatronic(player: Player): Boolean = animatronics.any { it.player == player }
}
